eng/pipelines/common/maui-templates.yml

```diff
@@ -55,69 +55,76 @@ parameters:
-        Write-Host "Current job status is: $env:AGENT_JOBSTATUS"
-        if ($env:AGENT_JOBSTATUS -eq "SucceededWithIssues") {
-            Write-Host "##vso[task.complete result=Failed;]DONE"
-        }
-      displayName: Fail if any issues occurred
+- job: build_maui_templates
+  workspace:
+    clean: all
+  displayName: 'Build platform:'
+  timeoutInMinutes: 240
+  condition: ${{ parameters.condition}}
+  strategy:
+    matrix:
+      ${{ each BuildPlatform in parameters.BuildPlatforms }}:
+        ${{ BuildPlatform.name }}:
+          POOL_NAME: ${{ BuildPlatform.poolName }}
+          POOL_VIMAGE: ${{ BuildPlatform.vmImage }}
+          PLATFORM_NAME: ${{ lower(BuildPlatform.name) }}
+
+  pool:
+    name: $(POOL_NAME)
+    vmImage: $(POOL_VIMAGE)
+    demands:
+      - macOS.Name -equals Ventura
+      - macOS.Architecture -equals x64
+  steps:
+
+  - ${{ each step in parameters.prepareSteps }}:
+    - ${{ each pair in step }}:
+        ${{ pair.key }}: ${{ pair.value }}
+
+  - template: provision.yml
+    parameters:
+      platform: ${{ variables.PLATFORM_NAME }}
+      checkoutDirectory: ${{ parameters.checkoutDirectory }}
+
+  - task: DownloadBuildArtifacts@0
+    displayName: 'Download Packages'
+    inputs:
+      artifactName: ${{ parameters.artifactName }}
+      itemPattern: ${{ parameters.artifactItemPattern }}
+      downloadPath: $(System.DefaultWorkingDirectory)/artifacts
+
+  - pwsh: Move-Item -Path artifacts\${{ parameters.artifactName }}\*.nupkg -Destination artifacts -Force
+    displayName: Move the downloaded artifacts
+
+  - pwsh: ./build.ps1 --target=dotnet-local-workloads --verbosity=diagnostic
+    displayName: 'Install .NET (Local Workloads)'
+    retryCountOnTaskFailure: 3
+    workingDirectory: ${{ parameters.checkoutDirectory }}
+    env:
+      DOTNET_TOKEN: $(dotnetbuilds-internal-container-read-token)
+      PRIVATE_BUILD: $(PrivateBuild)
+
+  - task: DotNetCoreCLI@2
+    inputs:
+      projects: ${{ parameters.checkoutDirectory }}/src/TestUtils/src/Microsoft.Maui.IntegrationTests/Microsoft.Maui.IntegrationTests.csproj
+    displayName: Build Microsoft.Maui.IntegrationTests
+
+  - task: DotNetCoreCLI@2
+    inputs:
+      command: test
+      projects: ${{ parameters.checkoutDirectory }}/src/TestUtils/src/Microsoft.Maui.IntegrationTests/bin/Debug/net7.0/Microsoft.Maui.IntegrationTests.dll
+      arguments: --logger "console;verbosity=normal" --filter "FullyQualifiedName=Microsoft.Maui.IntegrationTests.TemplateTests"
+      publishTestResults: true
+      testRunTitle: $(PLATFORM_NAME) template build tests
+    displayName: $(PLATFORM_NAME) template build tests
+    continueOnError: true
+
+  - pwsh: |
+      Write-Host "Current job status is: $env:AGENT_JOBSTATUS"
+      if ($env:AGENT_JOBSTATUS -eq "SucceededWithIssues") {
+          Write-Host "##vso[task.complete result=Failed;]DONE"
+      }
+    displayName: Fail if any issues occurred
 
 - ${{ each RunPlatform in parameters.RunPlatforms }}:
```

Comment:
Should we do the same for the run templates jobs? I think we'd want to use the matrix strategy here as well if it produces less YAML but that could be done in a future PR.

Reply:
You are 100% correct, I was going to dos it in a future PR :)